packages/engine-server/src/engineClient.ts

```diff
@@ -318,8 +324,15 @@ export class DendronEngineClient implements DEngineClient {
           );
         }
         this.notes[id] = ent.note;
+        this.noteFnames.add(ent.note);
       }
     });
+    // There's a limitation with updates: we know what the new note is, but we don't know what the old note was.
+    // If the note was renamed, then we can't remove the old name from `noteFnames` because we don't know the old fname.
+    // To get around this, we'll just refresh the entire dict.
+    if (notes.some((note) => note.status === "update")) {
```

Comment:
will this be expensive? its called every time the note changes (eg. [onDidChangeTextDocument](https://github.com/dendronhq/dendron/blob/9b58d335bcaa86692dfef895f15b5e48220e962b/packages/plugin-core/src/WorkspaceWatcher.ts#L160) gets called everytime the note changes)

can we update `NoteChangeEntry` signature to the following?
```ts
export type NoteChangeEntry = {
  notePrev: NoteProps;
  noteNext: NoteProps;
  status: "create" | "update" | "delete";
};
```

this is a pre-requisite to a bunch of later link optimizations (currently we calculate all links on every update but this lets us be more selective depending on what props have changed)